### PR TITLE
Save shoot list filter options in local storage

### LINKED ISSUE #866: Filter prefences are lost on page refresh
**What happened**:
If I change the filters on  https://dashboard.example.com/namespace/_all/shoots/ after refres (ctrl+R/F5) the filters are reset to the default one.

**What you expected to happen**:
The filters settings to be persisted and to survive page refresh.

**How to reproduce it (as minimally and precisely as possible)**:
Open the https://dashboard.example.com/namespace/_all/shoots/ page, change some of the filters settings, refresh the page and see that the filters are reset to the default values.

**Anything else we need to know**:

**Environment**:
Dashboard Version: 1.46.1
Browser: Firefox 82.0 on Ubuntu 18.04

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
Save shoot list filter options in local storage

**Which issue(s) this PR fixes**:
Fixes #866 

**Special notes for your reviewer**:

**Release note**:

```improvement user

```

### CODE REVIEW

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -132,8 +131,11 @@ class Shortcut {
   }
 }
 
-const getFilterValue = (state) => {
-  return state.namespace === '_all' && state.onlyShootsWithIssues ? 'issues' : null
+const getFilterValue = (getters) => {
+  if (state.namespace === '_all' && getters.onlyShootsWithIssues) {
+    return 'issues'
+  }
+  return null
```
Comment (holgerkoser) [inline frontend/src/store/index.js:138]:
```suggestion
function getFilterValue (state, getters) {
  if (state.namespace === '_all' && getters.onlyShootsWithIssues) {
    return 'issues'
  }
  return null
```

Comment (petersutter) [reply frontend/src/store/index.js]: ✅

**Review by grolu — APPROVED** [review]
/lgtm

File: frontend/src/store/modules/shoots.js [hunk 2]
```diff
@@ -90,6 +90,9 @@ const getters = {
   getShootListFilters () {
     return state.shootListFilters
   },
+  onlyShootsWithIssues (state, getters) {
```
Comment (holgerkoser) [inline frontend/src/store/modules/shoots.js:93]: genau richtig 👏

File: frontend/src/views/ShootList.vue [hunk 3]
```diff
@@ -314,9 +314,12 @@ export default {
       }
     },
     toggleFilter (key) {
-      if (this.showOnlyShootsWithIssues) {
-        const filters = this.getShootListFilters
-        this.setShootListFilter({ filter: key, value: !filters[key] })
+      const filters = this.getShootListFilters
+      this.setShootListFilter({ filter: key, value: !filters[key] })
+      this.$localStorage.setObject('shootListFilter', pick(filters, ['onlyShootsWithIssues', 'progressing', 'userIssues', 'deactivatedReconciliation', 'hideTicketsWithLabel']))
```
Comment (holgerkoser) [inline frontend/src/views/ShootList.vue:319]: I don't understand why this works? Is the correct value stored?

File: frontend/src/store/index.js [hunk 4]
```diff
@@ -1094,17 +1099,19 @@ const actions = {
...
       })
   },
-  setShootListFilters ({ dispatch, commit }, value) {
-    return dispatch('shoots/setShootListFilters', value)
-      .catch(err => {
-        dispatch('setError', err)
-      })
+  async setShootListFilters ({ dispatch, getters }, value) {
+    try {
+      await dispatch('shoots/setShootListFilters', value)
```
Comment (holgerkoser) [inline frontend/src/store/index.js:1104]: I know it is no part of this PR but we don't need an action here. We could directly call the mutation in the shoot module.
```
  commit('shoots/SET_SHOOT_LIST_FILTERS', value)
```

Comment (petersutter) [reply frontend/src/store/index.js:1104]: `shoots/SET_SHOOT_LIST_FILTERS` is working with the rootState.
`modules/shoots.js`:
```js
  SET_SHOOT_LIST_FILTERS (state, { rootState, value }) {
    state.shootListFilters = value
    setFilteredAndSortedItems(state, rootState)
  },
```

 Not sure if I may pass state as rootState like e.g.
```js
commit('shoots/SET_SHOOT_LIST_FILTERS', { rootState: state, value })
```

Comment (holgerkoser) [reply frontend/src/store/index.js:1104]: You are right. We need at least the action on the root module. As I said, it not part of this PR. We will refactor that anyway and may get ride of the rootState at all. You can already remove the actions in the shoots module or keep them.

Comment (petersutter) [reply frontend/src/store/index.js:1104]: as discussed offline we will address this in the upcoming, planned refactoring

File: frontend/src/views/ShootList.vue (hunk 3 above)
Comment (petersutter) [reply frontend/src/views/ShootList.vue]: maybe it's now clearer?

**Pull request merged**